src/harness/reference_models/common/data.py

```diff
@@ -265,13 +265,16 @@ def getAuthorizedGrantsFromDomainProxies(domain_proxies):
   Returns:
     A list of |CbsdGrantInfo| for each authorized grant in the given Domain Proxies.
   """
-  grants = []
+  grant_records = []
+  cbsd_records = []
   for domain_proxy in domain_proxies:
     for cbsd in domain_proxy.getCbsdsWithAtLeastOneAuthorizedGrant():
+      reg_record = cbsd.getRegistrationRequest()
       for grant in cbsd.getAuthorizedGrants():
-        grants.append(
-            constructCbsdGrantInfo(
-                cbsd.getRegistrationRequest(),
-                grant.getGrantRequest()))
-
-  return grants
+        grant_record = grant.getGrantRequest()
+        grant_records.append(grant_record)
+      cbsd_records.append({
+                   'registration':reg_record,
+                   'grants': grant_records
+                   })
+  return cbsd_records
```

Comment:
This isn't the fix we want as it will break other usage. I think the issue is a bit deeper than it initially looks. I'll have a look into it a bit more thoroughly and discuss what the expected behavior is supposed to be with Kate. (Should have a PR up by the end of the day, hopefully sooner). In the meantime I'd be happy to approve the `WINNF_FT_S_MCP_testcase.py` changes as a separate PR (Minus the section Kate commented on).

Reply:
Thanks, restored this back to "master" copy and wait for your fix.